### PR TITLE
Add documentation for Testing

### PR DESCRIPTION
**Pull request by Emiller88**
@KevinMenden I'll let you add how to use all of your `nf-core modules` tools when they get merged.

### CODE REVIEW

**Comment by KevinMenden** [comment]
Yep sure I'll put something together at some point. Need to write docs for the actual `nf-core tools` repo anyway, because we don't have documentation about `nf-core modules ...` yet

File: README.md [hunk 1]
```diff
@@ -228,15 +228,28 @@ We also use a standardised parameter called `params.publish_dir_mode` that can b
...
+- Generic files from [`tests/data/`](tests/data/) MUST be reused by importing them as `file(${launchDir}/tests/data/fastq/rna/test_single_end.fastq.gz)`
 
-- Any outputs produced by the test workflow MUST be placed in a folder called [`test/output/`](software/fastqc/test/output/) so that they can be used for unit testing.
+- Any outputs produced by the test workflow MUST be included in the [pytest-workflow](https://pytest-workflow.readthedocs.io/en/stable) for that tool. md5sum is preferred, however it's acceptable to not have it on files that the hash changes due to various headers and timestamps (html). Please do your best to avoid just checking for the file being present.
 
 - If the appropriate test data doesn't exist for your module then it MUST be added to [`tests/data/`](tests/data/).
 
-- A GitHub Actions workflow file MUST be added to [`.github/workflows/`](.github/workflows/) e.g. [`.github/workflows/fastqc.yml`](.github/workflows/fastqc.yml).
+- A filter for the module must be created in [`.github/filters.yml`](.github/filters.yml). Please include any pathes specific that tool or upstream of that tool (For example bowtie build is upstream of bowtie align).
```
Comment (KevinMenden) [inline README.md:239]:
```suggestion
- A filter for the module must be created in [`.github/filters.yml`](.github/filters.yml). Please include any paths specific for that tool or upstream of that tool (For example bowtie build is upstream of bowtie align).
```

File: README.md [hunk 2]
```diff
@@ -228,15 +228,28 @@ We also use a standardised parameter called `params.publish_dir_mode` that can b
 
 - Test files MUST be kept as tiny as possible.
 
-- Every module MUST be tested by adding a test workflow with a toy dataset in the [`test/`](software/fastqc/test) directory of the module.
+- Every module MUST be tested by adding a test workflow with a toy dataset in the [`tests/`](tests/software/fastqc/main.nf) test directory of the module.
```
Comment (KevinMenden) [inline README.md:231]:
```suggestion
- Every module MUST be tested by adding a test workflow with a toy dataset in the [`tests/`](tests/software/fastqc/main.nf) directory of the module.
```

`tests test directory' maybe a bit too much test

**Review by KevinMenden — APPROVED** [review]
LGTM.
Just a typo and a small style thing that can be added or ignored as desired :) 
👍

**Comment by drpatelh** [comment]
Thanks a bunch @Emiller88 ! Will merge and maybe add some stuff about software builds too.

**Pull request merged**